### PR TITLE
Replaced sprintf by MPL_snprintf

### PR DESCRIPTION
**Pull request by tarudoodi**

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by raffenet** [comment]
@tarudoodi this fails to compile because of an undeclared variable with ch4/ucx. Can you fix and push as an additional commit? That way I will only re-test the ucx job and squash before merging. Thanks.

**Comment by tarudoodi** [comment]
Pushed the changes as additional commit.

**Comment by raffenet** [comment]
test:jenkins/ch4/ucx

File: src/mpi/romio/adio/ad_lustre/ad_lustre_open.c [hunk 1]
```diff
@@ -57,6 +57,7 @@ void ADIOI_LUSTRE_Open(ADIO_File fd, int *error_code)
     lum = (struct lov_user_md *)ADIOI_Calloc(1,lumlen);
 
      value = (char *) ADIOI_Malloc((MPI_MAX_INFO_VAL+1)*sizeof(char));
+    int value_sz = (MPI_MAX_INFO_VAL+1)*sizeof(char);
```
Comment (raffenet) [inline src/mpi/romio/adio/ad_lustre/ad_lustre_open.c]: can you move this variable declaration up with the others, and also make it a `size_t`? Then we can avoid doing the size calculation here and also in `ADIOI_Malloc`

File: src/mpi/romio/test/async-multiple.c [hunk 2]
```diff
@@ -35,6 +35,7 @@ int main(int argc, char **argv)
     MPI_Status status[NR_NBOPS];
     MPI_Request request[NR_NBOPS];
     int errcode = 0;
+    int filename_sz;
```
Comment (raffenet) [inline src/mpi/romio/test/async-multiple.c]: Use `size_t`.

**Comment by tarudoodi** [comment]
Submitted the changes as a new commit.

**Comment by raffenet** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Pull request merged**